### PR TITLE
feat(scalasdk): add standard dependencies to project via sbt plugin.

### PR DESCRIPTION
**Pull request by RayRoestenburg**

### CODE REVIEW

**Review by raboof — APPROVED** [review]
Seems reasonable to me

File: samples/scala-value-entity-customer-registry/src/test/scala/customer/domain/CustomerValueEntitySpec.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/* This code was generated by Akka Serverless tooling.
```
Comment (johanandren) [inline samples/scala-value-entity-customer-registry/src/test/scala/customer/domain/CustomerValueEntitySpec.scala:1]: This looks unrelated though?

**Review by johanandren — APPROVED** [review]

File: samples/scala-value-entity-customer-registry/src/test/scala/customer/domain/CustomerValueEntitySpec.scala (hunk 1 above)
Comment (RayRoestenburg) [reply samples/scala-value-entity-customer-registry/src/test/scala/customer/domain/CustomerValueEntitySpec.scala:1]: yeah that was me running a test, I'll remove it from the PR.

**Comment by RayRoestenburg** [comment]
I added some settings to exclude the standard added dependencies, maybe overkill.

File: samples/scala-value-entity-customer-registry/src/test/scala/customer/domain/CustomerValueEntitySpec.scala (hunk 1 above)
Comment (RayRoestenburg) [reply samples/scala-value-entity-customer-registry/src/test/scala/customer/domain/CustomerValueEntitySpec.scala]: removed.

**Comment by johanandren** [comment]
Can't removing them be done by a user anyway without additional settings here? `libraryDependencies := libraryDedendencies.value.filter()`

File: sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala [hunk 2]
```diff
@@ -41,14 +48,39 @@ object AkkaserverlessPlugin extends AutoPlugin {
...
+    val excludeScalaTestDependencies = settingKey[Boolean]("Exclude scalatest dependencies")
   }
   object autoImport extends Keys
   import autoImport._
 
+  val LogbackVersion = "1.2.3"
+  val LogbackContribVersion = "0.1.5"
+  val logback = "ch.qos.logback" % "logback-classic" % LogbackVersion
+  val logbackJson = "ch.qos.logback.contrib" % "logback-json-classic" % LogbackContribVersion
+  val logbackJackson = "ch.qos.logback.contrib" % "logback-jackson" % LogbackContribVersion
```
Comment (patriknw) [inline sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala:61]: I don't think the Logback dependencies should be included by the plugin. Instead they should be included as ordinary transitive dependencies via the sdk library. I thought I had already fixed that in https://github.com/lightbend/akkaserverless-java-sdk/pull/496 ?

File: sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala [hunk 3]
```diff
@@ -41,14 +48,39 @@ object AkkaserverlessPlugin extends AutoPlugin {
...
+  val ScalaTestVersion = "3.2.7"
```
Comment (patriknw) [inline sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala:62]: I'm leaning towards not including ScalaTest dependency because we all know there are opinions about test frameworks and by having that as an explicit dependency in the example build it's easier to switch. We only depend on it in the unmanaged (one time) generated test class so users can replace that code with their own test framework code.

We have a similar approach for Java where users can switch between Junit4 and 5 or whatnot.

**Review by patriknw — COMMENTED** [review]
not sure this is the right approach...

File: sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala (hunk 2 above)
Comment (RayRoestenburg) [reply sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala:61]: Ah, did not know that.. yes is fine.

File: sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala (hunk 3 above)
Comment (RayRoestenburg) [reply sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala:62]: ok

File: samples/scala-value-entity-customer-registry/build.sbt [hunk 4]
```diff
@@ -28,20 +28,3 @@ Test / logBuffered := false
...
-val logbackJson = "ch.qos.logback.contrib" % "logback-json-classic" % LogbackContribVersion
-val logbackJackson = "ch.qos.logback.contrib" % "logback-jackson" % LogbackContribVersion
-
-// FIXME include these dependencies via the AkkaserverlessPlugin
-val AkkaServerlessSdkVersion = System.getProperty("akkaserverless-sdk.version", "0.7.2")
-libraryDependencies ++= Seq(
-  logback,
-  logbackJson,
-  logbackJackson,
-  "org.scalatest" %% "scalatest" % "3.2.7" % Test,
```
Comment (patriknw) [inline samples/scala-value-entity-customer-registry/build.sbt:44]: ScalaTest must remain here, if you agree with https://github.com/lightbend/akkaserverless-java-sdk/pull/508#discussion_r717260609

File: sbt-plugin/src/sbt-test/sbt-akkaserverless/basic/build.sbt [hunk 5]
```diff
@@ -1,9 +1,3 @@
 scalaVersion := "2.13.6"
 
 enablePlugins(AkkaserverlessPlugin)
-
-val AkkaServerlessSdkVersion = System.getProperty("plugin.version", "0.7.2")
-libraryDependencies ++= Seq(
-  "org.scalatest" %% "scalatest" % "3.2.7" % Test,
```
Comment (patriknw) [inline sbt-plugin/src/sbt-test/sbt-akkaserverless/basic/build.sbt:7]: ScalaTest must remain here, if you agree with https://github.com/lightbend/akkaserverless-java-sdk/pull/508#discussion_r717260609

File: samples/scala-value-entity-customer-registry/build.sbt [hunk 6]
```diff
@@ -28,20 +28,3 @@ Test / logBuffered := false
...
-  // FIXME include testkit dependency via the AkkaserverlessPlugin
-  "com.akkaserverless" %% "akkaserverless-scala-sdk-testkit" % AkkaServerlessSdkVersion % Test)
-
```
Comment (patriknw) [inline samples/scala-value-entity-customer-registry/build.sbt:47]: don't we have another Scala sample? Fibonacci? Is that up to date?

File: sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala (hunk 2 above)
Comment (RayRoestenburg) [reply sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala]: removed these.

File: sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala (hunk 3 above)
Comment (RayRoestenburg) [reply sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala]: removed.

File: samples/scala-value-entity-customer-registry/build.sbt (hunk 6 above)
Comment (RayRoestenburg) [reply samples/scala-value-entity-customer-registry/build.sbt:47]: ah good catch

File: sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala (hunk 3 above)
Comment (RayRoestenburg) [reply sbt-plugin/src/main/scala/com/akkaserverless/sbt/AkkaserverlessPlugin.scala]: (yeah this is why I added the settings to remove the deps but yeah that was just an idea)
